lib/livebook_web/live/session_live.ex

```diff
@@ -2272,10 +2272,16 @@ defmodule LivebookWeb.SessionLive do
   end
 
   defp is_secret_on_session?(secret, secrets) do
-    Map.has_key?(secrets, secret.name)
+    Map.has_key?(secrets, secret.name) && secrets[secret.name] == secret.value
   end
 
   defp get_saved_secrets do
     Enum.sort(Hubs.get_secrets() ++ Secrets.get_secrets())
   end
+
+  defp secret_label(%{origin: {:hub, id}}), do: Hubs.fetch_hub!(id).hub_emoji
+  defp secret_label(_), do: nil
+
+  defp secret_tooltip(%{origin: {:hub, id}}), do: Hubs.fetch_hub!(id).hub_name
```

Comment:
I asked this already, but do we remove all secrets of a given hub when the hub is deleted? Otherwise deleting a hub and keeping its secrets will make this crash.

Reply:
However, at this point I wonder if we should load the hub when we load the secret. I.e., let's make it be `{:hub, %Hub{}}` so we don't need to load data in a loop elsewhere? This code is effectively a `N+1` query and, even though it is ETS, we should probably avoid it?

Reply:
When we terminate the child, it doesn't have the secrets anymore. I can add the `:connection` subscription and implements the broadcast from `EnterpriseClient.stop/1`, so we can receive the event and get the saved secrets again, removing all the secrets from deleted hub.

Reply:
Ah, you are right, they are never stored. I think there is a simpler solution. I believe we already load all `@hubs` when the sidebar loads, so let's pass the hubs to the component and look for them inside the `@hubs` list, instead of from storage. We need to make sure that we also add and remove hubs from the `@hubs` list as they are added and removed.

Reply:
Basically, whenever a hub is added or deleted, we fetch all hubs AND all hubs secrets.